clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3207,9 +3207,21 @@ static void checkDimensionsAndSetDiagnostics(Sema &S, FunctionDecl *New,
   if (!NewDeclAttr || !OldDeclAttr)
     return;
 
-  if ((NewDeclAttr->getXDim() != OldDeclAttr->getXDim()) ||
-      (NewDeclAttr->getYDim() != OldDeclAttr->getYDim()) ||
-      (NewDeclAttr->getZDim() != OldDeclAttr->getZDim())) {
+  int64_t NewXDimVal =
+      NewDeclAttr->getXDim()->getIntegerConstantExpr(S.Context)->getSExtValue();
```

Comment:
Should this be unsigned?

Reply:
Yes, Thanks @elizabethandrews for looking into the patch. I have updated the values.

Reply:
Done